test/loopback-dl.test.js

```diff
@@ -837,6 +837,83 @@ describe('Models attached to a dataSource', function() {
 
   });
 
+  context('replaceAttributes', function() {
+    it('should replace with new data', function(done) {
+      Post.create({title: 'a', content: 'AAA', comments: ['Comment1']},
+      function(err, post) {
+        if (err) return done(err);
+        post.replaceAttributes({title: 'b'}, function(err, p) {
+          if (err) return done(err);
+          p.id.should.equal(post.id);
+          p.should.not.have.property('_id');
+          p.title.should.equal('b');
+          p.should.not.have.property('content');
+          p.should.not.have.property('comments');
+          Post.findById(post.id, function(err, p) {
+            if (err) return done(err);
+            p.id.should.equal(post.id);
+            p.should.not.have.property('_id');
+            p.title.should.equal('b');
+            // should.equal(p.content, undefined);
+            // should.equal(p.comments, undefined);
```

Comment:
I would prefer `p.should.have.property('content', undefined)` or `p.should.not.have.property('content')`, depending on what is actually expected.